components/script/dom/webglrenderingcontext.rs

```diff
@@ -1323,25 +1343,46 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
     }
 
     // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.10
+    fn DisableVertexAttribArray(&self, attrib_id: u32) {
+        if attrib_id > self.limits.max_vertex_attribs {
+            return self.webgl_error(InvalidValue);
+        }
+
+        self.ipc_renderer
+            .send(CanvasMsg::WebGL(WebGLCommand::DisableVertexAttribArray(attrib_id)))
+            .unwrap()
+    }
+
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.10
     fn GetActiveUniform(&self, program: Option<&WebGLProgram>, index: u32) -> Option<Root<WebGLActiveInfo>> {
-        program.and_then(|p| match p.get_active_uniform(index) {
-            Ok(ret) => Some(ret),
-            Err(error) => {
-                self.webgl_error(error);
-                None
-            },
-        })
+        if let Some(program) = program {
```

Comment:
I think here (and below too) will be cleaner like:

```rust
let program = match program {
    Some(program) => program,
    None => {
        self.webgl_error(InvalidValue);
        return None;
     }
};

// And so on.
```

That being said, I can't see what backs on the spec generating an `InvalidValue` error if there's no program, can you find a source?